src/vm/comdelegate.cpp

```diff
@@ -1076,8 +1076,20 @@ PCODE COMDelegate::ConvertToCallback(MethodDesc* pMD)
     if (NDirect::MarshalingRequired(pMD, pMD->GetSig(), pMD->GetModule()))
         COMPlusThrow(kNotSupportedException, W("NotSupported_NonBlittableTypes"));
 
-    // Get UMEntryThunk from appdomain thunkcache cache.
-    UMEntryThunk *pUMEntryThunk = GetAppDomain()->GetUMEntryThunkCache()->GetUMEntryThunk(pMD);
+    UMEntryThunkCache* pCache;
+    LoaderAllocator* pLoaderAllocator = pMD->GetLoaderAllocator();
+    if (pLoaderAllocator->IsCollectible())
```

Comment:
Is there a reason for this check - why can't be just do `GetLoaderAllocator()->GetUMEntryThunkCache()` unconditionally?

Reply:
Hmm, interesting. So you mean we would then remove the cache from the AppDomain and have it only in the LoaderAllocator. That sounds fine. Although we could possibly end up with more instances of the cache, the overhead would be negligible.

Reply:
Right.

> we could possibly end up with more instances of the cache

I do not think we would end up with more instances of the cache.